src/api/FactoryAPI.ts

```diff
@@ -105,6 +110,41 @@ export class FactoryAPI {
     return extractNewSetTokenAddressFromLogs(transactionLogs);
   }
 
+  /**
+   * Calculates the minimum allowable natural unit for a list of ERC20 component addresses
+   * where the minimum natural unit allowed equal 10^(18-minDecimal)
+   *
+   * @param componentAddresses    Component ERC20 addresses
+   *
+   * @return                      The minimum value of the natural unit allowed by component decimals
+   */
+  public async calculateMinimumNaturalUnit(
+    components: Address[],
+  ): Promise<BigNumber> {
+    const componentInstancePromises = _.map(components, component => {
+      return DetailedERC20Contract.at(component, this.web3, {});
+    });
+
+    const componentInstances = await Promise.all(componentInstancePromises);
+
+    let minDecimal;
+    try {
+      const decimalPromises = _.map(componentInstances, componentInstance => {
```

Comment:
Sometimes we use lodash `map` and other time we just do `array`.map. Is there a difference?